pkg/spinup/spinup.go

```diff
@@ -0,0 +1,64 @@
+package spinup
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	BaseURL      = "http://localhost:8090"
+	SpaceURI     = "/api/v2/spaces"
+	ResourceURI  = "/api/v2/resources"
+	ServerURI    = "/api/v2/servers"
+	SizeURI      = "/api/v2/sizes"
+	ContainerURI = "/api/v2/containers"
+	StorageURI   = "/api/v2/storage"
+)
+
+// FlexInt is an int... or a string... or an int.... or...
+type FlexInt int
+
+// Client is the spinup client
+type Client struct {
+	HTTPClient *http.Client
+}
+
+// ResourceType is an interface for deteriming URLs
+type ResourceType interface {
+	GetEndpoint(id string) string
+}
+
+// Resource gets details about a resource
```

Comment:
`... and unmarshals them them into the passed ResourceType`
It wasn't immediately clear what this does since it just returns an error